dramatiq/composition.py

```diff
@@ -45,8 +52,9 @@ def __init__(self, children, *, broker=None):
             else:
                 messages.append(child.copy())
 
-        for message, next_message in zip(messages, messages[1:]):
-            message.options["pipe_target"] = next_message.asdict()
+        rev = messages[::-1]
```

Comment:
Is this change necessary?

Reply:
Yes, it is. `asdict` returns a copy of the message content. If you set `pipe_target` going from first to last, the message inside of it won't have its own `pipe_target`. You can try it yourself: remove this line and run the tests, there is one test (I don't remember the name) that reproduces it.

Reply:
I see, thanks! So that means the earlier change to `asdict` was technically a change in behaviour. I prefer this new behaviour, but I'll have to think about whether or not it might cause issues downstream.

Reply:
It happens because dataclasses.asdict creates a copy of collections:

![image](https://user-images.githubusercontent.com/9638362/209776921-8cea8cae-e16d-4706-a055-2a530d456447.png)

I think we can preserve the old behavior if we explicitly put the real value of `options` instead of the copy on top of the copied result.

Reply:
> I think we can preserve the old behavior if we explicitly put the real value of options instead of the copy on top of the copied result.

Yes, I think we should do that.

Reply:
Done. I made `Message.asdict` not to create copies and removed reverse loop from `pipeline.__init__`